### DISCUSSION TITLE
Handling complexity in scala-maven-plugin code

### DISCUSSION THREAD

**Discussion by slandelle** [davidB/scala-maven-plugin#660] [Ideas]
Hey there,

As one of the 2 active committers, I feel like I have to deal with lots of extra luggage and complexity when introducing changes (I don't pretend to know everything then).

I see 2 main issues.

## Dead features

I'm not convinced some features are actually used:
* continuous compilation
* continuous test
* scala console
* help?
* run
* script

I'd be in favor of dropping these.

## Code too open

In the past, I tried to harden visibility (private package) and extensibility (final). This change broke some metals plugin for maven because they were extending this plugin. IMHO, this is very wrong to me and a maven plugin is not something that should be extended by a third party. On the contrary, I would completely agree with metals contributing upstream to this repository, as long as they contribute on maintaining these features.

WDYT? It would be great to have feedback from users.

**Comment by jozic** [comment]
Agree on both points. Not even sure what  some of those features are and i use the plugin for years.
